python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -241,20 +249,33 @@ def _update(self):
             return
 
         self.last_update_time = now
-        self.update_worker_list()
+
+        # Query node provider to update workers, all_workers, and all_nodes.
+        self.update_node_lists()
 
         # Remove from LoadMetrics the ips unknown to the NodeProvider.
-        self.load_metrics.prune_active_ips([
-            self.provider.internal_ip(node_id) for node_id in self.all_workers
-        ])
+        self.load_metrics.prune_active_ips(
+            [self.provider.internal_ip(node_id) for node_id in self.all_nodes])
+
+        # Update status strings
+        logger.info(self.info_string())
+        legacy_log_info_string(self, self.workers)
```

Comment:
what is the difference between self.workers and self.all_workers?

Reply:
let's call them managed_workers to make it clearer

Reply:
I think what happened originally was that there were workers and then the "unmanaged workers" concept was invented so "all workers = workers + unmanaged_workers" was added.

Reply:
If we just delete the "unmanaged worker" concept from code base, probably no one will notice... :p

Reply:
This has been refactored.